common/achievement/index.ts

```diff
@@ -127,11 +127,12 @@ function injectRecordValue(condition: string, recordValue: number) {
     return condition;
 }
 
-async function awardUser(evaluationResult: number, userAchievement: UserAchievementTemplate) {
+async function rewardUser<ID extends ActionID>(evaluationResult: number, userAchievement: UserAchievementTemplate, event: ActionEvent<ID>) {
     let newRecordValue = null;
     if (typeof userAchievement.recordValue === 'number' && evaluationResult) {
         newRecordValue = evaluationResult;
     }
+    await actionTakenAt(new Date(event.at), event.user, 'reward_issued', event.context);
```

Comment:
Can we call the `actionTakenAt()` function after the update? Because if in any case the update does not work, we still send the notification...

Reply:
and also here: maybe the context will change